willsCheatMenu/src/hacks/misc.ts

```diff
@@ -2,70 +2,132 @@ import { Swal, Toast, NumberInput, Confirm } from "../utils/swal";
+      _.player.equipment.setHat(rand(gameData.hat));
+      _.player.appearanceChanged = true;
+    }, 690);
+  })
+  .setDisabled(() => {
+    if (viber) clearInterval(viber);
+  });
 
 new Hack(category.misc, "Bobbify", "Converts your account into Bobby Fancywoman.").setClick(async () => {
-	if (!(
-		await Confirm.fire("Are you sure you want your account to be turned into Bobby Fancywoman?", "This action is not reversable.")
-	).value) return;
-
-	_.player.name.data.nickname = null;
-	_.player.name.data.firstName = 44;
-	_.player.name.data.middleName = 754;
-	_.player.name.data.lastName = 882;
-	_.player.data.stars = -1e22;
-	_.player.data.level = 69;
-
-	_.player.appearance.setGender("male");
-	_.player.appearance.setEyeColor(1);
-	_.player.appearance.setFace(4);
-	_.player.appearance.setHair(19, 1);
-	_.player.appearance.setSkinColor(1);
-	_.player.equipment.setFollow(19);
-	_.player.equipment.setHat(19);
-	_.player.equipment.setBoots(19);
-	_.player.equipment.setOutfit(19);
-	_.player.equipment.setWeapon(19);
-	_.player.forceSaveCharacter();
-	await Toast.fire("Bobbified!", "You are now Bobby Fancywoman.", "success");
+  if (!(
+      await Confirm.fire("Are you sure you want your account to be turned into Bobby Fancywoman?", "This action is not reversable.")
+    ).value) return;
+
+  _.player.name.data.nickname = null;
+  _.player.name.data.firstName = 44;
+  _.player.name.data.middleName = 754;
+  _.player.name.data.lastName = 882;
+  _.player.data.stars = -1e22;
+  _.player.data.level = 69;
+
+  _.player.appearance.setGender("male");
+  _.player.appearance.setEyeColor(1);
+  _.player.appearance.setFace(4);
+  _.player.appearance.setHair(19, 1);
+  _.player.appearance.setSkinColor(1);
+  _.player.equipment.setFollow(19);
+  _.player.equipment.setHat(19);
+  _.player.equipment.setBoots(19);
+  _.player.equipment.setOutfit(19);
+  _.player.equipment.setWeapon(19);
+  _.player.forceSaveCharacter();
+  await Toast.fire("Bobbified!", "You are now Bobby Fancywoman.", "success");
 });
 
-new Hack(category.misc, "Reset Account","Completely resets your account.").setClick(async () => {
-	if (!(await Confirm.fire("Are you sure you want to reset your account?", "This action is not reversable.")).value) return;
-	_.player.resetAccount();
-});
\ No newline at end of file
+new Hack(category.misc, "Reset Account", "Completely resets your account.").setClick(async () => {
+  if (!(await Confirm.fire("Are you sure you want to reset your account?", "This action is not reversable.")).value) return;
+  _.player.resetAccount();
+});
+
+new Hack(category.misc, "Restore Player State", "Restores your player data to an earlier state (WCM must've been installed at the time, and the history data is local to this device only.)").setClick(async () => {
+  playerStateHistory = JSON.parse(localStorage.getItem("playerStateHistory")) || [];
+  
+  inputOptionsPromise = new Promise(function(resolve) {
+    setTimeout(function() {
```

Comment:
Can I have more info about why you used setTimeout?

Reply:
i copied something from stackoverflow and it had a `setTimeout` thing